client/my-sites/plugins/with-plugin-redirect.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useRef, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslate, translate as __ } from 'i18n-calypso';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { isJetpackSite, getSiteWoocommerceWizardUrl } from 'calypso/state/sites/selectors';
+import {
+	isPluginActionCompleted,
+	isPluginActionInProgress,
+} from 'calypso/state/plugins/installed/selectors';
+import { INSTALL_PLUGIN } from 'calypso/lib/plugins/constants';
+import getSiteConnectionStatus from 'calypso/state/selectors/get-site-connection-status';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import { successNotice, removeNotice } from 'calypso/state/notices/actions';
+
+// List of plugins that should perform a redirect
+// once it's installed.
+const redirectingPluginsList = {
```

Comment:
We could put those directly in the component. That way we could use the `translate()` from `useTranslate` directly.

Reply:
sounds goot to me

Reply:
Actually, I think it's ok keeping these data out of the component. I think it looks clearer and also allows us to move these data to another location, in case we want to do it.

Reply:
moved 😅